frontend/src/LineGraph.res

```diff
@@ -48,65 +48,33 @@ let computeStats = xs => {
 module DataRow = {
   type name = string
   type units = string
-  type value
-  type metric = {name, value, units}
-  type t = array<value>
+  type value = float
+  type metric = {name: name, value: value, units: units}
+  type t = array<value> // Currently array of length 3: [min, value, max]
+  // The type of each row in the data supplied to Dygraph: [index, dataArray, statsArray]
+  type row = array<t>
 
-  let single = (x: float): value =>
-    Obj.magic([Obj.magic(Js.null), Obj.magic(x), Obj.magic(Js.null)])
+  external makeFloat: 'a => float = "%identity"
+  let floatNull = makeFloat(Js.null)
```

Comment:
Can we use [Js.Nullable](https://rescript-lang.org/docs/manual/latest/api/js/nullable) to get rid of this last unsafe bit? (or does it cause some issue with the js lib?)

Reply:
I don't understand exactly what you mean here, when you ask about using `Js.Nullable`. Could you elaborate a little more? `DataRow.t` is an array of floats but the js lib accepts nulls and so, we use this `floatNull` where required.

Reply:
Yes sorry the rescript documentation is not very good. I found it via https://rescript-lang.org/docs/manual/latest/null-undefined-option where they talk a bit about the representation of `'a option` for interrop with javascript... but it's not exactly what we want as it uses `undefined` rather than `null` for `None`, so they point to `Js.Nullable` as an alternative solution.

My understanding is that a `Js.Nullable.t<float>` will be either a null or a js float at runtime (using `return x` for `Some x`). This would allow for a type safe representation of an array of floats/null that should play well with the dygraph library... but I haven't tried it!

Reply:
I've pushed a commit to move all the magic code to a single place - before we pass data onto the dygraph library. @art-w let me know if this looks reasonable to you.